ttnn/cpp/ttnn/operations/moreh/moreh_adam/moreh_adam.hpp

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/moreh/moreh_adam/device/moreh_adam_device_operation.hpp"
```

Comment:
This include is not needed here, right?

Reply:
I've removed it. It was there so that the .cpp could share the same includes.

Reply:
@o2buzzle here are my general thoughts on this

### Avoid unnecessary includes for shorter build/rebuild times
If your .hpp does not require anything from an included header, it should be removed. 
.hpp is a contract with which other .hpp/.cpp files work. 
any time a .cpp changes, you need to rebuild only that object file. 
but anytime an .hpp changes, you need to rebuild anything that includes it.

### Avoid leaking details of implementation
Operation can be implemented in many different ways.
Whatever is in its header is now public - a detail known to any consumer.
Anything defined in that header can now be used and relied upon, restricting you in future refactors.